lib/tbd/psi.rb

```diff
@@ -435,157 +468,254 @@ def gen(id = "")
+      v[:joint       ] = 0; v[:transition         ] = 0
+
+      v[:joint              ] = @set[id][:joint              ] if h[:joint              ]
+      v[:transition         ] = @set[id][:transition         ] if h[:transition         ]
+      v[:fenestration       ] = @set[id][:fenestration       ] if h[:fenestration       ]
+      v[:head               ] = @set[id][:fenestration       ] if h[:fenestration       ]
+      v[:headconcave        ] = @set[id][:fenestration       ] if h[:fenestration       ]
+      v[:headconvex         ] = @set[id][:fenestration       ] if h[:fenestration       ]
+      v[:sill               ] = @set[id][:fenestration       ] if h[:fenestration       ]
+      v[:sillconcave        ] = @set[id][:fenestration       ] if h[:fenestration       ]
+      v[:sillconvex         ] = @set[id][:fenestration       ] if h[:fenestration       ]
+      v[:jamb               ] = @set[id][:fenestration       ] if h[:fenestration       ]
+      v[:jambconcave        ] = @set[id][:fenestration       ] if h[:fenestration       ]
+      v[:jambconvex         ] = @set[id][:fenestration       ] if h[:fenestration       ]
+      v[:door               ] = @set[id][:fenestration       ] if h[:fenestration       ]
+      v[:doorhead           ] = @set[id][:fenestration       ] if h[:fenestration       ]
+      v[:doorheadconcave    ] = @set[id][:fenestration       ] if h[:fenestration       ]
+      v[:doorheadconvex     ] = @set[id][:fenestration       ] if h[:fenestration       ]
+      v[:doorsill           ] = @set[id][:fenestration       ] if h[:fenestration       ]
+      v[:doorsillconcave    ] = @set[id][:fenestration       ] if h[:fenestration       ]
+      v[:doorsillconvex     ] = @set[id][:fenestration       ] if h[:fenestration       ]
+      v[:doorjamb           ] = @set[id][:fenestration       ] if h[:fenestration       ]
+      v[:doorjambconcave    ] = @set[id][:fenestration       ] if h[:fenestration       ]
+      v[:doorjambconvex     ] = @set[id][:fenestration       ] if h[:fenestration       ]
+      v[:skylight           ] = @set[id][:fenestration       ] if h[:fenestration       ]
+      v[:skylighthead       ] = @set[id][:fenestration       ] if h[:fenestration       ]
+      v[:skylightheadconcave] = @set[id][:fenestration       ] if h[:fenestration       ]
+      v[:skylightheadconvex ] = @set[id][:fenestration       ] if h[:fenestration       ]
+      v[:skylightsill       ] = @set[id][:fenestration       ] if h[:fenestration       ]
+      v[:skylightsillconcave] = @set[id][:fenestration       ] if h[:fenestration       ]
+      v[:skylightsillconvex ] = @set[id][:fenestration       ] if h[:fenestration       ]
+      v[:skylightjamb       ] = @set[id][:fenestration       ] if h[:fenestration       ]
+      v[:skylightjambconcave] = @set[id][:fenestration       ] if h[:fenestration       ]
+      v[:skylightjambconvex ] = @set[id][:fenestration       ] if h[:fenestration       ]
```

Comment:
By default, all subsurface perimeter edges inherit from **:fenestration** PSI factor entries. If **:door** and/or **:skylight** entries are found in a given PSI factor set, they are used instead. This ensures (to a large degree) backwards compatibility with previous tests and use cases.